rasa_sdk/executor.py

```diff
@@ -77,38 +97,43 @@ def utter_button_template(
         **kwargs: Any,
     ) -> None:
         """Sends a message template with buttons to the output channel."""
+        warnings.warn(
+            "Use of `utter_button_template` is deprecated. "
+            "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
 
-        message = {"template": template, "buttons": buttons}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        self.utter_message(template=template, buttons=buttons)
 
     # noinspection PyUnusedLocal
     def utter_template(
         self, template: Text, tracker: Tracker, silent_fail: bool = False, **kwargs: Any
     ) -> None:
         """"Send a message to the client based on a template."""
+        warnings.warn(
+            "Use of `utter_template` is deprecated. " "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
 
-        message = {"template": template}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        self.utter_message(template=template)
 
     def utter_custom_json(self, json_message: Dict[Text, Any], **kwargs: Any) -> None:
         """Sends custom json to the output channel."""
+        warnings.warn(
+            "Use of `utter_custom_json` is deprecated. " "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
 
-        json_message = {"custom": json_message}
-        json_message.update(kwargs)
-
-        self.messages.append(json_message)
+        self.utter_message(json_message=json_message)
 
     def utter_image_url(self, image: Text, **kwargs: Any) -> None:
         """Sends url of image attachment to the output channel."""
+        warnings.warn(
+            "Use of `utter_image_url` is deprecated. " "Use `utter_message` instead.",
```

Comment:
```suggestion
            "Use of `utter_image_url` is deprecated. Use `utter_message` instead.",
```

Reply:
Thanks for these. Sorry you had to do this 😓, I will start checking strings after black runs :/